src/transformStateWithClones.js

```diff
@@ -8,6 +8,31 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const copyState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(copyState, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (let i = 0; i < action.keysToRemove.length; i++) {
+        delete copyState[action.keysToRemove[i]];
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (const key of Object.keys(copyState)) {
+        if (copyState[key]) {
+          delete copyState[key];
+        }
+      }
+    }
```

Comment:
[CODE STYLE]: Use a switch statement if you have a limited amount of conditions. This will make your code cleaner and more maintainable. Replace the if statements with a switch statement.